File: src/main/java/com/refinedmods/refinedstorage/container/transfer/ItemHandlerInventoryWrapper.java
package com.refinedmods.refinedstorage.container.transfer;

import net.minecraft.world.item.ItemStack;
import net.minecraftforge.items.IItemHandler;
import net.minecraftforge.items.ItemHandlerHelper;

import java.util.Objects;

class ItemHandlerInventoryWrapper implements IInventoryWrapper {
    private final IItemHandler handler;

    ItemHandlerInventoryWrapper(IItemHandler handler) {
        this.handler = handler;
    }

    @Override
    public InsertionResult insert(ItemStack stack) {
        return new InsertionResult(ItemHandlerHelper.insertItem(handler, stack, false));
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }

        if (o == null || getClass() != o.getClass()) {
            return false;
        }

        ItemHandlerInventoryWrapper that = (ItemHandlerInventoryWrapper) o;

        return Objects.equals(handler, that.handler);
    }

    @Override
    public int hashCode() {
        return Objects.hash(handler);
    }
}

File: src/main/kotlin/com/refinedmods/refinedstorage/container/transfer/ItemHandlerInventoryWrapper.kt
package com.refinedmods.refinedstorage.container.transfer

import com.refinedmods.refinedstorage.api.network.grid.IGrid.slotId
import net.minecraft.item.ItemStack
import net.minecraftforge.items.IItemHandler
import net.minecraftforge.items.ItemHandlerHelper
import java.util.*

internal class ItemHandlerInventoryWrapper(handler: IItemHandler?) : IInventoryWrapper {
    private val handler: IItemHandler?
    override fun insert(stack: ItemStack?): InsertionResult? {
        return InsertionResult(ItemHandlerHelper.insertItem(handler, stack, false))
    }

    override fun equals(o: Any?): Boolean {
        if (this === o) {
            return true
        }
        if (o == null || javaClass != o.javaClass) {
            return false
        }
        val that = o as ItemHandlerInventoryWrapper
        return handler == that.handler
    }

    override fun hashCode(): Int {
        return Objects.hash(handler)
    }

    init {
        this.handler = handler
    }
}